src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -37,59 +310,488 @@ const ActorInit En_Zo_InitVars = {
+    Vec3f surfacePos;
+    f32 yDist;
+    f32 hDist;
+
+    surfacePos.x = this->actor.posRot.pos.x;
+    surfacePos.y = this->actor.posRot.pos.y + this->actor.waterY;
+    surfacePos.z = this->actor.posRot.pos.z;
+
+    hDist = Math_Vec3f_DistXZ(&surfacePos, &player->actor.posRot.pos);
+    yDist = fabsf(player->actor.posRot.pos.y - surfacePos.y);
+
+    if (hDist < 240.0f && yDist < 80.0f) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnZo_SetAnimation(EnZo* this) {
+    s32 animID = 8;
+
+    if (((this->skelAnime.animation == &D_06000598) || (this->skelAnime.animation == &D_06000D48))) {
+        if (this->unk_194.unk_00 == 0) {
+            if (this->actionFunc == EnZo_Standing) {
+                animID = 0;
+            } else {
+                animID = 3;
+            }
+        }
+    }
+
+    if ((this->unk_194.unk_00 != 0) && (this->actor.textId == 0x4006) && (this->skelAnime.animation != &D_06000598)) {
+        animID = 6;
+    }
+
+    if ((this->unk_194.unk_00 != 0) && (this->actor.textId == 0x4007) && (this->skelAnime.animation != &D_06000D48)) {
+        animID = 7;
+    }
+
+    if (animID != 8) {
+        func_80034EC0(&this->skelAnime, sAnimations, animID);
+        if (animID == 3) {
+            this->skelAnime.animCurrentFrame = this->skelAnime.animFrameCount;
+            this->skelAnime.animPlaybackSpeed = 0.0f;
+        }
+    }
+}
+
+void EnZo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600BFA8, NULL, this->limbDrawTbl, this->transitionDrawTbl, 20);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    func_80061EFC(&this->actor.colChkInfo, NULL, &sColChkInit);
+
+    if (LINK_IS_ADULT && ((this->actor.params & 0x3F) == 8)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    func_80034EC0(&this->skelAnime, sAnimations, 2);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.unk_1F = 6;
+    this->dialogRadius = this->collider.dim.radius + 30.0f;
+    this->unk_64C = 1;
+    this->canSpeak = false;
+    this->unk_194.unk_00 = 0;
+    func_8002E4B4(globalCtx, &this->actor, this->collider.dim.height * 0.5f, this->collider.dim.radius, 0.0f, 5);
+
+    if (this->actor.waterY < 54.0f || (this->actor.params & 0x3F) == 8) {
+        this->actor.shape.shadowDrawFunc = ActorShadow_DrawFunc_Circle;
+        this->actor.shape.unk_10 = 24.0f;
+        func_80034EC0(&this->skelAnime, sAnimations, 1);
+        this->canSpeak = true;
+        this->alpha = 255.0f;
+        this->actionFunc = EnZo_Standing;
+    } else {
+        this->actor.flags &= ~1;
+        this->actionFunc = EnZo_Submerged;
+    }
+}
+
+void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnZo_Standing(EnZo* this, GlobalContext* globalCtx) {
+    s16 angle;
```

Comment:
we arent too focused on documentation stuff at this time, but just want to mention anyway, for action functions we try to name them as an action the actor is currently doing. So id personally do `EnZo_Stand` and maybe `EnZo_StayUnderWater` or something, that one is kinda hard to keep in the scheme.


Surface/TreadWater/Dive are perfect tho

Reply:
I kinda like `Standing` and `Submerged` because those actions are more passive, if that makes any sense.
maybe it's just a me thing lol,

Reply:
yea i can see that. i prefer them present tense when possible, but by all means keep it that way for now. we are far from being able to standardize documentation stuff like this, so itll be revisited at a later time